// unstructured-client/dist/commonjs/lib/config.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SDK_METADATA = exports.serverURLFromOptions = exports.ServerList = exports.ServerDevelopment = exports.ServerSaasApi = void 0;
const url_js_1 = require("./url.js");
/**
 * Serverless SaaS API
 */
exports.ServerSaasApi = "saas-api";
/**
 * Development server
 */
exports.ServerDevelopment = "development";
/**
 * Contains the list of servers available to the SDK
 */
exports.ServerList = {
    [exports.ServerSaasApi]: "https://api.unstructuredapp.io",
    [exports.ServerDevelopment]: "http://localhost:8000",
};
function serverURLFromOptions(options) {
    let serverURL = options.serverURL;
    const params = {};
    if (!serverURL) {
        const server = options.server ?? exports.ServerSaasApi;
        serverURL = exports.ServerList[server] || "";
    }
    const u = (0, url_js_1.pathToFunc)(serverURL)(params);
    return new URL(u);
}
exports.serverURLFromOptions = serverURLFromOptions;
exports.SDK_METADATA = {
    language: "typescript",
    openapiDocVersion: "1.1.7",
    sdkVersion: "0.24.1",
    genVersion: "2.568.2",
    userAgent: "speakeasy-sdk/typescript 0.24.1 2.568.2 1.1.7 unstructured-client",
};
